python/ray/train/huggingface/accelerate/accelerate_trainer.py

```diff
@@ -0,0 +1,422 @@
+    See full example code below.
+
+    .. testcode::
+
+        def train_loop_per_worker():
+            # Report intermediate results for callbacks or logging and
+            # checkpoint data.
+            session.report(...)
+
+            # Get dict of last saved checkpoint.
+            session.get_checkpoint()
+
+            # Session returns the Ray Dataset shard for the given key.
+            session.get_dataset_shard("my_dataset")
+
+            # Get the total number of workers executing training.
+            session.get_world_size()
+
+            # Get the rank of this worker.
+            session.get_world_rank()
+
+            # Get the rank of the worker on the current node.
+            session.get_local_rank()
+
+    For more information, see the documentation of
+    :class:`~ray.train.torch.TorchTrainer`.
+
+    .. note::
+
+        You need to use ``session.report()`` to communicate results and checkpoints
+        back to Ray Train.
+
+    Accelerate integrations with DeepSpeed, FSDP, MegatronLM etc. are fully supported.
+    If the Accelerate configuration contains a path to a DeepSpeed config file
+    (``deepspeed_config_file``), that file will also be loaded and applied on the
+    workers.
+
+    The following Accelerate configuration options will be ignored and automatically
+    set by the Trainer according to Ray AIR configs (eg. ``ScalingConfig``):
+    - Number of machines (``num_machines``)
+    - Number of processes (``num_processes``)
+    - Rank of the current machine (``machine_rank``)
+    - Local rank of the current machine
+    - GPU IDs (``gpu_ids``)
+    - Number of CPU threads per process (``num_cpu_threads_per_process``)
+    - IP of the head process (``main_process_ip``)
+    - Port of the head process (``main_process_port``)
+    - Whether all machines are on the same network (``same_network``)
+    - Whether to force a CPU-only mode (``cpu``/``use_cpu``)
+    - rdzv backend (``rdzv_backend``)
+    - Main training function (``main_training_function``)
+    - Type of launcher
+
+    This Trainer requires ``accelerate>=0.17.0`` package.
+    It is tested with ``accelerate==0.17.1``.
+
+    Example:
+        .. testcode::
+
+            import torch
+            import torch.nn as nn
+
+            from accelerate import Accelerator
+
+            import ray
+            from ray.air import session, Checkpoint
+            from ray.train.huggingface.accelerate import AccelerateTrainer
+            from ray.air.config import ScalingConfig
+            from ray.air.config import RunConfig
+            from ray.air.config import CheckpointConfig
+
+            # If using GPUs, set this to True.
+            use_gpu = False
+
+            # Define NN layers archicture, epochs, and number of workers
+            input_size = 1
+            layer_size = 32
+            output_size = 1
+            num_epochs = 200
+            num_workers = 3
```

Comment:
same comment here. can we not duplicate this code?

Reply:
This is the same as in the TorchTrainer docstring

Reply:
come in and update all of them all at once?